### DISCUSSION TITLE
HW 3 Part 4

### DISCUSSION THREAD

**Discussion by ecieszynski** [cooperrc/computational-mechanics#55] [Q&A] [answered]
Kind of confused if this is the right way to go about starting this problem, because the plot doesn't seem correct. Any guidance would be appreciated

![image](https://user-images.githubusercontent.com/106542680/174823630-65597346-ffa9-4d7c-8aae-3dcecd65c5ab.png)

**Comment by cooperrc** [comment, marked as answer]
use `lambda` here

```python
sig_T = lambda T: np.exp(a0-a1*T))/(1+np.exp(a0-a1*T))
T = np.linspace(0, 100)
plt.plot(T, sig_T(T))
```

That will open the function, `sig_T` up for root-finding rather than assigning it to an array

**Reply by ecieszynski** [reply]
can you briefly explain the lambda function for me. Saw you used it in an example from the videos and it was talked about in the jupyter notebooks as well, but I am still not sure what it does and when to use it

**Reply by cooperrc** [reply]
`lambda` is a quick function definition method. It _can_ have many of the same functionality as `def`, but it usually doesn't have docstrings or multiple optional parameters. 

for example the `sincos(x)` function could be

```python
def sincos(x):
    return np.sin(x), np.cos(x)
```

or

```python
sincos = lambda x: (np.sin(x), np.cos(x))
```

**Comment by caroline-cook** [comment]
I'm not exactly sure why your format didn't work, as I understand why you wrote it out the way you did. But what I did, which made sense to me, was that instead of making T an arange, I used the "for i in range" function. Instead of T in the function, I used temperature[I]. Then you can just plot sigmoid instead of T, sig_T, since it's already a function of temperature. You should get a different plot. I hope this helps!
temperature = []
for i in range(101):
    temperature.append(i) 
sigmoid = []
for i in range(101): 
    sigmoid.append(insert function)
